p2p/libp2p/discovery/hostDecorator.go

```diff
@@ -0,0 +1,182 @@
+package discovery
+
+import (
+	"context"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/libp2p/go-libp2p-core/connmgr"
+	"github.com/libp2p/go-libp2p-core/event"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	protocol "github.com/libp2p/go-libp2p-core/protocol"
+	peerstore "github.com/libp2p/go-libp2p-peerstore"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+const (
+	hardCPSLimitLow  = 2
+	hardCPSLimitHigh = 100
+)
+
+type hostDecorator struct {
+	h            host.Host
```

Comment:
Why doesn't hostDecorator directly extend host.Host? All funcs beyond line 114 might be deleted + corresponding test `TestDecoratorCallRelay`.

Reply:
done